tasks/deploy/coordinators/shared.ts

```diff
@@ -0,0 +1,61 @@
+import { subtask, types } from "hardhat/config";
+import dayjs from "dayjs";
+import duration from "dayjs/plugin/duration";
+import { validHyperdrivePrefixes } from "../utils";
+dayjs.extend(duration);
+
+export type DeployCoordinatorsBaseParams = {
+  prefix: keyof typeof validHyperdrivePrefixes;
+};
+
+/**
+ * Deploys the core deployer and all targets for a deployer coordinator
+ * - Unfortunately, by parameterizing the contract names we lose some type-checking ability
+ */
+subtask(
+  "deploy:coordinators:shared",
+  "shared deployment steps for deployer coordinators",
+)
+  .addParam(
+    "prefix",
+    "contract prefix for the coordinator",
+    undefined,
+    types.string,
+  )
+  .setAction(
+    async (
+      { prefix }: DeployCoordinatorsBaseParams,
+      { deployments, run, viem },
+    ) => {
+      let prefixValue = validHyperdrivePrefixes[prefix];
+      // Deploy the core deployer
+      console.log(`deploying ${prefixValue}HyperdriveCoreDeployer...`);
+      let core = await viem.deployContract(
+        `${prefixValue}HyperdriveCoreDeployer` as any,
+      );
+      await deployments.save(`${prefixValue}HyperdriveCoreDeployer`, {
+        ...core,
+        args: [],
+      } as any);
+      await run("deploy:verify", {
+        name: `${prefixValue}HyperdriveCoreDeployer`,
+      });
+
+      // Deploy the targets
+      let targets: `0x${string}`[] = [];
+      for (let i = 0; i < 5; i++) {
```

Comment:
Just a heads up that this will get bumped to 6 in the k-inflation PRs. I was thinking about something today, and this reminds me. We should add a function to `HyperdriveDeployerCoordinator` that records how many targets need to be deployed for a full deployment. That would be helpful to be able to parameterize loops like this without magic numbers.

I can add this once we get the k-inflation PR in, but I didn't want to forget about it.

Reply:
Thanks for the call out! Hoping the switch should be pretty easy